test/unit/core/RegistryUnit.t.sol

```diff
@@ -0,0 +1,356 @@
+    }
+
+    function test_AcceptProfileOwnershipWhenCalledWithCorrectParams(bytes32 _profileId) external {
+        address _pendingOwner = makeAddr("pendingOwner");
+        stdstore.target(address(registry)).sig("profileIdToPendingOwner(bytes32)").with_key(_profileId).checked_write(
+            _pendingOwner
+        );
+
+        // it should emit ProfileOwnerUpdated
+        vm.expectEmit(true, true, true, true);
+        emit ProfileOwnerUpdated(_profileId, _pendingOwner);
+
+        vm.prank(_pendingOwner);
+        registry.acceptProfileOwnership(_profileId);
+
+        // it should delete the pendingOwner from profileIdToPendingOwner
+        assertEq(registry.profileIdToPendingOwner(_profileId), address(0));
+    }
+
+    function test_AddMembersRevertWhen_MemberIsZeroAddress(bytes32 _profileId)
+        external
+        givenCallerIsProfileOwner(_profileId)
+    {
+        address[] memory _members = new address[](1);
+        _members[0] = address(0);
+
+        // it should revert
+        vm.expectRevert(Errors.ZERO_ADDRESS.selector);
+
+        registry.addMembers(_profileId, _members);
+    }
+
+    function test_AddMembersWhenCalledWithCorrectParams(bytes32 _profileId)
+        external
+        givenCallerIsProfileOwner(_profileId)
+    {
+        address[] memory _members = new address[](1);
+        _members[0] = makeAddr("member");
+        // it should call _grantRole for members
+        registry.expectCall__grantRole(_profileId, _members[0]);
+
+        registry.addMembers(_profileId, _members);
+    }
+
+    function test_RemoveMembersWhenCalled(bytes32 _profileId) external givenCallerIsProfileOwner(_profileId) {
+        address[] memory _members = new address[](1);
+        _members[0] = makeAddr("member");
+
+        // it should call _revokeRole for members
+        registry.expectCall__revokeRole(_profileId, _members[0]);
+
+        registry.removeMembers(_profileId, _members);
+    }
+
+    function test__checkOnlyProfileOwnerRevertWhen_CallerIsNotProfileOwner(bytes32 _profileId) external {
+        // it should revert
+        vm.expectRevert(Errors.UNAUTHORIZED.selector);
+
+        registry.call__checkOnlyProfileOwner(_profileId);
+    }
+
+    function test__checkOnlyProfileOwnerWhenCallerIsProfileOwner(bytes32 _profileId) external {
+        registry.mock_call__isOwnerOfProfile(_profileId, address(this), true);
+
+        // it should not revert
+        registry.call__checkOnlyProfileOwner(_profileId);
+    }
+
+    function test__generateProfileIdShouldReturnTheKeccak256OfEncodedNonceAndOwnerAddress(
+        uint256 _nonce,
+        address _owner
+    ) external {
+        // it should return the keccak256 of encoded nonce and owner address
+        assertEq(keccak256(abi.encodePacked(_nonce, _owner)), registry.call__generateProfileId(_nonce, _owner));
+    }
+
+    function test__isOwnerOfProfileWhenProvidedAddressIsOwnerOfProfile(bytes32 _profileId, address _owner) external {
+        vm.skip(true);
```

Comment:
Is something wrong with the test?

Reply:
something wrong with stdstorage couldnt make it work for now

Reply:
You can add a TODO so we can fix it later